pkg/vault/contracts/WrappedBalancerPoolTokenFactory.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import {
+    IWrappedBalancerPoolTokenFactory
+} from "@balancer-labs/v3-interfaces/contracts/vault/IWrappedBalancerPoolTokenFactory.sol";
+
+import { WrappedBalancerPoolToken } from "./WrappedBalancerPoolToken.sol";
+
+/// @notice Factory contract for creating wrapped Balancer pool tokens
+contract WrappedBalancerPoolTokenFactory is IWrappedBalancerPoolTokenFactory {
+    IVault internal immutable _vault;
+    mapping(address => address) internal _wrappedTokens;
+
+    constructor(IVault vault_) {
+        _vault = vault_;
+    }
+
+    /// @inheritdoc IWrappedBalancerPoolTokenFactory
+    function createWrappedToken(address bpt) external returns (address) {
+        address wrappedToken = _wrappedTokens[bpt];
+        if (wrappedToken != address(0)) {
+            revert WrappedBPTAlreadyExists(wrappedToken);
+        }
```

Comment:
What happens if the factory needs to be deprecated/updated? I wonder if this register shouldn't stay in the registry contract.